sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/models/PhoneNumberErrorResponseException.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.phonenumbers.models;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpResponse;
+
+/**
+ * Exception thrown on phone number specific errors
+ */
+public class PhoneNumberErrorResponseException extends HttpResponseException {
```

Comment:
The Java guidelines don't want us to create custom exception types, can we just throw the HttpResponseException directly?

https://azure.github.io/azure-sdk/java_introduction.html#exceptions

Reply:
@srnagar Can you help us clarify what the right Java pattern is for converting custom service exceptions and throwing them?

Reply:
Is this a good pattern to model after?
https://github.com/Azure/azure-sdk-for-java/blob/9ea60ce411df334c136c48817e0be55d8e8b928e/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java#L217-L233

Reply:
But this is following the pattern established in Chat which passed arch board review already. If we want to change the pattern, we'd better change both phone number and chat.